roles/openshift_monitor_availability/files/web-console-smoke-test.yaml

```diff
@@ -0,0 +1,117 @@
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: web-console-smoke-test-template
+  annotations:
+    openshift.io/display-name: Web Console Smoke Test
+    iconClass: icon-openshift
+    tags: openshift,infra
+    # TODO: update this URL when we migrate up the repo
+    openshift.io/documentation-url: https://github.com/benjaminapetersen/origin-web-console-smoke-test
+    openshift.io/support-url: https://access.redhat.com
+    openshift.io/provider-display-name: Red Hat, Inc.
+    description: >
+      Runs smoke tests in a headless browser against the web console at a provided URL.
+
+      For more information about using this template, see
+      https://github.com/benjaminapetersen/origin-web-console-smoke-test/blob/master/README.md
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: web-console-smoke-test
+    namespace: ${NAMESPACE}
+    labels:
+      app: web-console-smoke-test
+  spec:
+    # replica set
+    replicas: ${REPLICA_COUNT}
+    selector:
+      matchLabels:
+        app: web-console-smoke-test
+    template:
+      metadata:
+        labels:
+          app: web-console-smoke-test
+      spec:
+        containers:
+        - name: web-console-smoke-test
+          image: ${IMAGE}
+          imagePullPolicy: ${PULL_POLICY}
+          env:
+          - name: TOKEN
+            value: ${TOKEN}
+          - name: CONSOLE_URL
+            value: ${CONSOLE_URL}
+          - name: CONSOLE_USER
+            value: ${CONSOLE_USER}
+          - name: CONSOLE_PASSWORD
+            value: ${CONSOLE_PASSWORD}
+          - name: TEST_INTERVAL_MINUTES
+            value: ${TEST_INTERVAL_MINUTES}
+          ports:
+          - containerPort: 3000
+          restartPolicy: Always
+          volumeMounts:
+            - mountPath: /etc/tls-certs
+              name: web-console-smoke-test-serving-cert
+              readOnly: true
+        volumes:
+        - name: web-console-smoke-test-serving-cert
+          secret:
+            defaultMode: 400
+            secretName: web-console-smoke-test-serving-cert
+
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: web-console-smoke-test
+    namespace: ${NAMESPACE}
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: web-console-smoke-test-serving-cert
+      prometheus.io/scrape: "true"
+      prometheus.io/scheme: https
```

Comment:
@ironcladlou We'll need guidance on the right way to get the results scraped.

Reply:
We'll need PRs to https://github.com/openshift/cluster-monitoring-operator for this; I need to make one for the monitor-app-create app too

Reply:
@ironcladlou not sure what kind of PR's you need to the https://github.com/openshift/cluster-monitoring-operator repo

Reply:
Sorry for the ambiguity... right now, at least, just adding the deployment and service for your app into the openshift-monitor-availability namespace isn't enough to get the app scraped by Prometheus. The role additions in this PR will get your app **installed into the cluster**, but to get the app scraped by Prometheus, you must [follow these instructions](https://github.com/openshift/cluster-monitoring-operator#contributing-new-component-integrations) to incorporate the component with the Cluster Monitoring Operator setup.

Reply:
@ironcladlou our app is a nodejs app, so we can't follow those instructions as written :(

Reply:
OK, I just spoke with @ironcladlou. It looks like we need to open a PR against the cluster monitoring operator repo with those changes. It's not something we need to change in our own image. The `ServiceMonitor` resource tells Prometheus what services to scrape.

@jhadvig Can you take a look? Please follow up with me or @ironcladlou if you have questions.

Reply:
@spadgett I think those instructions are pointing to changes in the openshift/cluster-monitoring-operator repository, not in our web console monitoring app. @ironcladlou correct ?

Reply:
Yes, for your app to be discovered/scraped by Prometheus, you must make a PR to the cluster-monitoring-operator repo to define a new ServiceMonitor pointing to your app's metrics service.